### PR TITLE
feat: Tokenless banners 

### PR DESCRIPTION
**Pull request by RulaKhaled**
# Description

- [x] https://github.com/codecov/codecov-api/pull/826

This updates tokeneless banners to display:
- Token required banner (admin view)
- Token required banner (member view)
- Token not required banner (admin view)
- Token not required banner (member view)

# Notable Changes
- New hook for tokenless
- Tokenless holds logic for required vs non required 
- Nested banners hold logic for admin vs member

# Screenshots
Screenshot 2024-10-06 at 1 48 09 PM
Screenshot 2024-10-06 at 1 48 16 PM
Screenshot 2024-10-06 at 1 39 26 PM
Screenshot 2024-10-06 at 1 39 17 PM

# Link to Sample Entry
Design: https://www.figma.com/design/CRM2ODN4ihBWITjC3xryF1/GH-2100?node-id=1-2&node-type=canvas&t=1YFaoET5ZEuqBoB4-0

# Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: src/services/uploadTokenRequired/useUploadTokenRequired.test.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            { wrapper }
+          )
+
+          await waitFor(() => expect(result.current.data).toBeNull())
+        })
+      })
+    })
+
+    describe('unsuccessful parse of zod schema', () => {
+      beforeEach(() => {
```
Comment (ajay-sentry) [inline src/services/uploadTokenRequired/useUploadTokenRequired.test.tsx:111]: nit: this and the other could be "beforeAll" and "afterAll."

File: src/shared/GlobalTopBanners/TokenlessBanner/TokenRequiredBanner/TokenRequiredBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        <p className="items-center gap-1 md:flex">
+          <span className="flex items-center gap-1 font-semibold">
+            <Icon name="informationCircle" />
+            You must now upload using a token.
+          </span>
+          Upload with either{' '}
+          {typeof orgUploadToken === 'string' ? (
+            <OrgUploadTokenTooltip orgUploadToken={orgUploadToken} />
+          ) : (
+            'the global upload token '
```
Comment (ajay-sentry) [inline src/shared/GlobalTopBanners/TokenlessBanner/TokenRequiredBanner/TokenRequiredBanner.tsx:93]: weird how it says this line is uncovered when you specifically have a test for it 🤔

**Review by ajay-sentry — APPROVED** [review]
nice work!

File: src/shared/GlobalTopBanners/TokenlessBanner/TokenRequiredBanner/TokenRequiredBanner.tsx (hunk 2 above)
Comment (RulaKhaled) [reply src/shared/GlobalTopBanners/TokenlessBanner/TokenRequiredBanner/TokenRequiredBanner.tsx:93]: it's been acting up like that lately i posed about it got no help :"D

**Pull request merged**